frontend/src/components/core/Sidebar/Sidebar.tsx

```diff
@@ -45,13 +47,36 @@ class Sidebar extends PureComponent<Props, State> {
   constructor(props: Props) {
     super(props)
 
+    // By default, expand sidebar only if browser width > MEDIUM_BREAKPOINT_PX
     const { innerWidth } = window || {}
 
     this.state = {
       collapsedSidebar: innerWidth
         ? innerWidth <= MEDIUM_BREAKPOINT_PX
         : false,
     }
+
+    // But if the app creator specified a sidebar state, use that instead.
+    if (this.props.initialSidebarState === PageConfig.SidebarState.EXPANDED) {
+      this.state = { collapsedSidebar: false }
+    }
+    if (this.props.initialSidebarState === PageConfig.SidebarState.COLLAPSED) {
+      this.state = { collapsedSidebar: true }
+    }
```

Comment:
Hm, the reason this isn't merged with `componentDidUpdate` is that I didn't want to introduce new sidebar behavior on app rerun, when `inital_sidebar_state` isn't set.

- On first run, default (AUTO) is that sidebar expands iff browser width > 1000
- On rerun (`componentDidUpdate`), default is that sidebar respects the current collapse state (if user collapsed, keep it collapsed)

I weakly think keeping this behavior is desirable, but could be persuaded to merge them and have the sidebar open/close on rerun.
----
Haha, there used to be an `if else` there but it was causing the line to exceed 80 chars and wrap, which looked ugly... I've just switched this entire thing to use a switch statement instead.

Reply:
I agree that not resetting would be desirable but then if the app developer changes the `inital_sidebar_state` they're making a change to the initial behavior that may contradict with what the user has done so who should be the one in control... I'm leaning more towards the user mostly because if the user has auto rerun, having the sidebar open/close automatically could be weird. I'm not strongly tied to this and can go either way.

Reply:
On further thought, I've gone ahead and merged the two together as suggested -- the corner case of the app developer changing the state to `AUTO` is both incredibly niche and arguably correct.